lib/Alchemy/Phrasea/SearchEngine/Elastic/ElasticSearchEngine.php

```diff
@@ -431,6 +417,42 @@ private function createRecordQueryParams($ESQuery, SearchEngineOptions $options,
         return $params;
     }
 
+    private function getAggregationQueryParams(SearchEngineOptions $options)
+    {
+        $params = [];
+
+        foreach ($options->getDataboxes() as $databox) {
+            foreach ($databox->get_meta_structure() as $fieldStructure) {
+                if (false === $fieldStructure->isAggregable()) {
+                    continue;
+                }
+
+                $indexField = 'caption';
+
+                if ($fieldStructure->isBusiness()) {
+                    if (false === $this->app['authentication']->isAuthenticated()) {
+                        continue;
+                    }
+                    $acl = $this->app['acl']->get($this->app['authentication']->getUser());
+                    if (false === $acl->can_see_business_fields($databox)) {
+                        continue;
+                    }
+
+                    $indexField = 'private_caption';
+                }
+
+                $params[$fieldStructure->get_name()] = [
```

Comment:
This is a **security issue**, this can give access to a private field to someone without the right to see it. Just have a databox with the same name and access to his private fields.